x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/logic/import/create_promise_from_rule_import_stream.test.ts

```diff
@@ -0,0 +1,385 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Readable } from 'stream';
+import { BadRequestError } from '@kbn/securitysolution-es-utils';
+
+import type { RuleToImport } from '../../../../../../common/api/detection_engine/rule_management';
+import {
+  getOutputDetailsSample,
+  getSampleDetailsAsNdjson,
+} from '../../../../../../common/api/detection_engine/rule_management/mocks';
+import type { InvestigationFields } from '../../../../../../common/api/detection_engine';
+import { createPromiseFromRuleImportStream } from './create_promise_from_rule_import_stream';
+
+export const getOutputSample = (): Partial<RuleToImport> => ({
+  rule_id: 'rule-1',
+  output_index: '.siem-signals',
+  risk_score: 50,
+  description: 'some description',
+  from: 'now-5m',
+  to: 'now',
+  index: ['index-1'],
+  name: 'some-name',
+  severity: 'low',
+  interval: '5m',
+  type: 'query',
+});
+
+export const getSampleAsNdjson = (sample: Partial<RuleToImport>): string => {
+  return `${JSON.stringify(sample)}\n`;
+};
+
+describe('createPromiseFromRuleImportStream', () => {
+  test('transforms an ndjson stream into a stream of rule objects', async () => {
+    const sample1 = getOutputSample();
+    const sample2 = getOutputSample();
+    sample2.rule_id = 'rule-2';
+    const ndJsonStream = new Readable({
+      read() {
+        this.push(getSampleAsNdjson(sample1));
+        this.push(getSampleAsNdjson(sample2));
+        this.push(null);
+      },
+    });
+    const [{ rules: result }] = await createPromiseFromRuleImportStream({
+      stream: ndJsonStream,
+      objectLimit: 1000,
+    });
+
+    expect(result).toEqual([
+      {
+        rule_id: 'rule-1',
+        output_index: '.siem-signals',
+        risk_score: 50,
+        description: 'some description',
+        from: 'now-5m',
+        to: 'now',
+        index: ['index-1'],
+        name: 'some-name',
+        severity: 'low',
+        interval: '5m',
+        type: 'query',
+      },
+      {
+        rule_id: 'rule-2',
+        output_index: '.siem-signals',
+        risk_score: 50,
+        description: 'some description',
+        from: 'now-5m',
+        to: 'now',
+        index: ['index-1'],
+        name: 'some-name',
+        severity: 'low',
+        interval: '5m',
+        type: 'query',
+      },
+    ]);
+  });
+
+  test('throws an error when the number of rules in the stream is equal to the limit', async () => {
```

Comment:
@yctercero FYI there was a TODO from you here that got moved as part of df456f24ae555c0e4ae6abeaff6b6745541e77a1.

BUT I ended up fixing the issue in 3cdbd3bff83ab8afc2e23d74ff707cfbfc6dbd8d, so the test is no longer skipped.

BUT I ended up observing (in the course of testing an edge case here) that we should probably either change the way we validate the object limit (e.g. swap `>=` for `>`), or change the message somehow.

Reply:
I would think the `>=` makes sense.